### DISCUSSION TITLE
Error loading schema when using a private constructor

### DISCUSSION THREAD

**Discussion by RomaKV** [ChilliCream/hotchocolate#4247] [Q&A] [answered]
The class Book contains two constructors. The first constructor is private with no parameters. The second constructor is public with parameters.
[https://github.com/RomaKV/TestHotChocolate](https://github.com/RomaKV/TestHotChocolate)
```
    public class Book
    {
        public int Id { get; private set; }
        public string Title { get; private set; }
        private Book(){}
        public Book(int id, string title)
        {
            Id = id;
            Title = title;
        }
    }
```
The schema cannot be reloaded. Continuously reported: "Retry schema fetching".

![image](https://user-images.githubusercontent.com/51952224/133940760-2f085d8b-741b-4db3-9ed8-d4a11aa7da4f.png)

 **Startup.cs**
```
public void ConfigureServices(IServiceCollection services)
        {
            services.AddControllersWithViews();
            services
                .AddGraphQLServer()
                .AddQueryType<BookType>()
                .AddMutationType<MutationType>();
        }
```
**BookType.cs**  
```
public class BookType : ObjectType<Book>
    {
        protected override void Configure(IObjectTypeDescriptor<Book> descriptor)
        {
            descriptor
                .Field(f => f.Title)
                .Type<StringType>();
            descriptor
                .Field(f => f.Id)
                .Type<IntType>();
        }
    }
```
**Mutation.cs**
```
    public class Mutation
    {
        public async Task<Book> AddBook(int id, string title)
        {
            return new Book(id, title);
        }
    }
```
**MutationType.cs**
```
public class MutationType : ObjectType<Mutation>
    {
        protected override void Configure(
            IObjectTypeDescriptor<Mutation> descriptor)
        {
            descriptor.Field(f => f.AddBook(default, default));
        }
    }
```
If the constructor Book(){} becomes public, the schema will be loaded successfully.
```
  public class Book
    {
        public int Id { get; private set; }
        public string Title { get; private set; }
        public Book(){}  // private doesn't work
        public Book(int id, string title)
        {
            Id = id;
            Title = title;
        }
    }
```
But, I can only use the private constructor no parameterless.
Why can't private constructors be used?
Please tell me how to get around this limit.

**Comment by michaelstaib** [comment, marked as answer]
The issue in your example is that you are using BookType as the query type.

The GraphQL root types must provide a public constructor or you can provide a custom DI registration in which case we will use your DI registration instead of looking for a constructor.

Why are you using book as a query type, this makes not really sense.
